server/src/data-processors/helpers/verify-auth.js

```diff
@@ -0,0 +1,42 @@
+const requiredDependency = require('../../global-helpers/required-dependency');
+const throwError = require('../../global-helpers/throw-error');
+
+function makeVerifyAuth({ jwt = requiredDependency('jwt') } = {}) {
+  const jwtParamHasVerifyMethod = 'verify' in jwt;
+  if (!jwtParamHasVerifyMethod) {
+    requiredDependency('jwt');
+  } 
+  else if (!process.env.AUTHSECRET) {
+    throwError('An "AUTHSECRET" environment variable is required', 'application');
+  } else {
+    return function (token) {
+      let message, tokenBody;
+  
+      if (token && typeof token === 'string') {
+        jwt.verify(token, process.env.AUTHSECRET, (err, decodedTokenBody) => {
+          if (err) {
+            throwError('You need authorisation to access this endpoint.', 'authorisation');
+          } else {
+            message = "Successfully authenticated.";
+            tokenBody = decodedTokenBody;
```

Comment:
Having a spot of trouble figuring out what form decodedTokenBody will have. Will it just be generic string?

Reply:
`decodedTokenBody` will be the payload object from the decoded token. Maybe I should rename it to `decodedTokenPayload`?